### ISSUE TITLE
NES clone list

### ISSUE THREAD

**Issue by AlvaroBurnett** [unexpectedpanda/retool#16]
Something strange happens after the latest clone list update for the NES.
Setting regions as default and language to English (only English), I get the following as "missing":

```
Adan y Eva (Spain) (Gluk Video) (Unl)
Bloque Magico, El (Spain) (Gluk Video) (Unl)
Booky Man (Brazil) (CCE) (Unl)
Booky Man (Spain) (Gluk Video) (Unl)
BugTris (Korea) (Unl)
Corre Benny (Spain) (Gluk Video) (Unl)
Creatom (Spain) (Gluk Video) (Unl)
Crime Busters (Brazil) (Unl)
Destructor, El (Spain) (Gluk Video) (Unl)
Gluk the Thunder Warrior (Spain) (Gluk Video) (Unl)
Gran Aventura Submarina, La (Spain) (Gluk Video) (Unl)
Janggun-ui Adeul (Korea) (Unl)
Koko Adventure (Korea) (Unl)
Magic Kid Googoo (Korea) (Unl)
Metal Force (Korea) (Unl)
Monstruo de los Globos, El (Spain) (Rev 1) (Gluk Video) (Unl)
Policeman (Spain) (Gluk Video) (Unl)
Skate Boy (Spain) (Gluk Video) (Unl)
War in the Gulf ~ Guerra del Golfo, La (Brazil, Spain) (CCE, Gluk Video) (Unl)
```

And the following as "unneeded":

```
Adam & Eve (Asia) (En) (Mega Soft) (Unl).zip
Go! Benny! (Asia) (En) (Unl).zip
Magic Block (Asia) (En) (Mega Soft) (Unl).zip

```

**Comment by unexpectedpanda** [comment]
There are a few different issues here.

I'm aware that Asian titles that should be parents are being demoted in favor of other regions. It's to do with Asia not having an implied language. Other regions without implied languages include Hong Kong, Taiwan, Latin America, Switzerland, South Africa, and Unknown, so in theory those could cause issues too, but in practice it only affects the NES dat and Asia for now. When I tried to fix this for 0.86, it had unintended consequences for the IBM dat, so I'll need to unravel a decent amount of the code I think. It's the top thing on my list to fix for 0.87.

So with things fixed in 0.87, and with English only set as a filter, it will probably look like this:

`Adam & Eve (Asia) (En) (Mega Soft) (Unl)` (kept) > `Adan y Eva (Spain) (Gluk Video) (Unl)` (removed)
`Koko Adventure (Korea) (Unl)` (kept) > `Buzz & Waldog` (USA) (Proto) (Unl)` (removed)
`Go! Benny! (Asia) (En) (Unl)` (kept) > `Corre Benny (Spain) (Gluk Video) (Unl)` (removed)
`Magic Block (Asia) (En) (Mega Soft) (Unl)` (kept) > `Bloque Magico, El (Spain) (Gluk Video) (Unl)` (removed)
`Gluk the Thunder Warrior (Spain) (Gluk Video) (Unl)` (kept) > `Thunder Warrior (Asia) (Ja) (Unl)` (removed)

Buzz & Waldog gets removed because it's a prototype, and Koko Adventure is production. Both Gluk the Thunder Warrior and Koko Adventure are listed in No-Intro's database as being English titles.

For Booky Man, potentially I'll need to add `(CCE)` as a publisher tag to make sure it gets auto-matched with the other Booky Man titles.

All remaining titles are listed as supporting English in No-Intro's database -- if that's incorrect, they'll need to change that.

I won't be able to get to this for a couple of days, but will look into it when I'm able.

**Comment by AlvaroBurnett** [comment]
Nice. It surprised me that this only affects unlicensed games. Don't know if that's just a coincidence.

One question: Do you know what's the best place to report No-Intro issues? I placed a message at their forum but I'm not sure if that's the best thing to do.

**Comment by unexpectedpanda** [comment]
The unlicensed code was the first place I looked, but at first glance it seems fine. It's looking like a coincidence, based on the code I've written so far.

The only place I'm aware of for No-Intro issues is their forums for the related platforms. From experience it doesn't seem like a response is guaranteed, sadly.

**Comment by unexpectedpanda** [comment]
This one opened up a can of worms, so it's taking longer to correct than expected. The fix will be out in the 0.87 release :)

**Comment by AlvaroBurnett** [comment]
Cool 👍 

Since No-Intro dats sometimes have incomplete language information, I'm thinking that it might be useful if Retool somehow allowed us to override language information for particular problematic roms.

**Comment by unexpectedpanda** [comment]
The maintenance burden for that is a step to far for what I'm willing to support at this stage. Keeping the clone lists up to date is an effort in itself!

**Comment by AlvaroBurnett** [comment]
Sure, I understand that, but it would be nice if the user could provide a file with that information via the command line.

**Comment by unexpectedpanda** [comment]
This is corrected in 0.87, which will be out shortly. Thanks for your patience!

I'm not going to pursue setting custom languages for titles -- however I've implemented custom filter lists where you can choose to force exclude or include titles, which should help you to achieve the same thing.

**Comment by AlvaroBurnett** [comment]
Awesome! That will be very useful.
